test/Index/index_module_refs.swift

```diff
@@ -0,0 +1,25 @@
+// RUN: %target-swift-ide-test -print-indexed-symbols -enable-source-import -source-filename %s -I %S/Store/Inputs | %FileCheck %s
+
+import ClangModuleB
+// CHECK: [[@LINE-1]]:8 | module/C | ClangModuleB | c:@M@ClangModuleB | Ref | rel: 0
+import ClangModuleC.Sub1
+// CHECK: [[@LINE-1]]:8 | module/C | ClangModuleC | [[ClangModuleC_USR:c:@M@ClangModuleC]] | Ref | rel: 0
+// CHECK: [[@LINE-2]]:21 | module/C | ClangModuleC.Sub1 | [[ClangModuleC_USR]]@M@Sub1 | Ref | rel: 0
+import ClangModuleC.Sub2
+// CHECK: [[@LINE-1]]:8 | module/C | ClangModuleC | [[ClangModuleC_USR]] | Ref | rel: 0
+// CHECK: [[@LINE-2]]:21 | module/C | ClangModuleC.Sub2 | [[ClangModuleC_USR]]@M@Sub2 | Ref | rel: 0
+import func ClangModuleA.funcA
+// CHECK: [[@LINE-1]]:13 | module/C | ClangModuleA | c:@M@ClangModuleA | Ref | rel: 0
+import SwiftModuleC
+// CHECK: [[@LINE-1]]:8 | module/Swift | SwiftModuleC | [[SwiftModuleC_USR:c:@M@SwiftModuleC]] | Ref | rel: 0
+
+struct MyType {}
+
+extension SwiftModuleC.MyType {
+  // CHECK: [[@LINE-1]]:11 | module/Swift | SwiftModuleC | [[SwiftModuleC_USR]] | Ref | rel: 0
+  func myMethod() {
+    _ = SwiftModuleC.MyGeneric<SwiftModuleC.MyType, MyType>()
+    // CHECK: [[@LINE-1]]:9 | module/Swift | SwiftModuleC | [[SwiftModuleC_USR]] | Ref | rel: 0
```

Comment:
One I think I noticed is that while the other type references have `| Ref,RelCont | rel: 1` (a 'RelCont' relation with the containing method), the module references do not have that.
I wouldn't say it is *super* important to have the containing relation, but could you take a look to see if it is simple to add, so that the module references do not diverge ?
If you see other test cases, we check for relations by using `CHECK-NEXT` to match the relations that are printed immediately after the symbol occurrence.

Reply:
Does this rule applies only to `AbstractFunctionDecl`? It seems that merely mimicking the generic Decl behaviour does the trick:

```
    auto Parent = getParentDecl();
    if (Parent && isa<AbstractFunctionDecl>(Parent))
      addRelation(Info, (unsigned)SymbolRole::RelationContainedBy, Parent);
```
`module/Swift | SwiftModuleC | [[SwiftModuleC_USR]] | Ref,RelCont | rel: 1`
` RelCont | instance-method/Swift | myMethod() | [[myMethod_USR:.*]]`